aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -782,6 +783,8 @@ def build(self, event: ResponseEventT, cors: Optional[CORSConfig] = None) -> Dic
             logger.debug("Encoding bytes response with base64")
             self.response.base64_encoded = True
             self.response.body = base64.b64encode(self.response.body).decode()
+        elif self.response.is_json():
+            self.response.body = self.serializer(self.response.body)
```

Comment:
```suggestion
            self.response.body = body if isinstance(self.response.body, str) else self.serializer(self.response.body)
```

suggestion so we can revert tests and don't accidentally double encode a JSON that would break a front-end